manifests/params.pp

```diff
@@ -6,6 +6,8 @@
 class virtualbox::params {
   $manage_ext_repo = true
   $repo_proxy      = undef
+  $apt_key_thumb   = undef
+  $apt_key_source  = undef
```

Comment:
Variables in Puppet are immutable (see: https://docs.puppet.com/puppet/latest/reference/lang_variables.html#no-reassignment). These variables are being assigned `undef` here, but are being reassigned to actual values below. IF you remove these two lines, your builds will go green (and I'll happily merge this PR).